sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala

```diff
@@ -1276,6 +1276,147 @@ class DataFrameWindowFunctionsSuite extends QueryTest
     )
   }
 
+  test("SPARK-46228: Insert window group limit node for cumulative aggregation with limit") {
+
+    val nullStr: String = null
+    val df = Seq(
+      ("a", 0, "c"),
+      ("a", 1, "x"),
+      ("a", 2, "y"),
+      ("a", 3, "z"),
+      ("a", 4, ""),
+      ("a", 4, ""),
+      ("b", 1, "h"),
+      ("b", 1, "n"),
+      ("c", 1, "z"),
+      ("c", 1, "a"),
+      ("c", 2, nullStr)).toDF("key", "value", "order")
+    val window = Window.partitionBy($"key").orderBy($"order".asc_nulls_first)
+    val window2 = Window.partitionBy($"key").orderBy($"order".desc_nulls_first)
+    val window3 = Window.orderBy($"order".asc_nulls_first)
+
+    Seq(true, false).foreach { enableEvaluator =>
+      withSQLConf(SQLConf.USE_PARTITION_EVALUATOR.key -> enableEvaluator.toString) {
+        Seq(-1, 100).foreach { threshold =>
+          withSQLConf(SQLConf.WINDOW_GROUP_LIMIT_THRESHOLD.key -> threshold.toString) {
+            // RowFrame
+            val existWindowGroupLimitRowFrame =
```

Comment:
We can move this one to sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/InferWindowGroupLimitSuite.scala

Reply:
Relevant tests have been added to `InferWindowGroupLimitSuite`, I will remove the redundant ones in `DataFrameWindowFunctionsSuite`.